src/IECoreArnold/ParameterAlgo.cpp

```diff
@@ -90,13 +90,37 @@ void setParameterInternal( AtNode *node, AtString name, int parameterType, bool
 		switch( parameterType )
 		{
 			case AI_TYPE_INT :
-				if( const IntData *data = dataCast<IntData>( name, value ) )
+				if( const Int64Data *data = runTimeCast<const Int64Data>( value ) )
+				{
+					int i = static_cast<int>( data->readable() );
+					if( i == data->readable() )
+					{
+						AiNodeSetInt( node, name, i );
+					}
+					else
+					{
+						msg( Msg::Warning, "setParameter", fmt::format( "Int64Data value {} is out of range for parameter \"{}\"", data->readable(), name ) );
+					}
+				}
+				else if( const IntData *data = dataCast<IntData>( name, value ) )
 				{
 					AiNodeSetInt( node, name, data->readable() );
 				}
 				break;
 			case AI_TYPE_UINT :
-				if( const IntData *data = runTimeCast<const IntData>( value ) )
+				if( const UInt64Data *data = runTimeCast<const UInt64Data>( value ) )
+				{
+					unsigned int i = static_cast<unsigned int>( data->readable() );
+					if( i == data->readable() )
+					{
+						AiNodeSetUInt( node, name, i );
+					}
+					else
+					{
+						msg( Msg::Warning, "setParameter", fmt::format( "UInt64Data value {} is out of range for parameter \"{}\"", data->readable(), name ) );
+					}
+				}
+				else if( const IntData *data = runTimeCast<const IntData>( value ) )
```

Comment:
The ordering of these cases feels weird to me - I'm used to seeing the most common case first in a list of branches like this ( it feels right from a performance point of view, though probably not actually meaningful for performance ... it's still ended up as the convention in my mind ). Regardless, I can't really come up a reason for the ordering UInt64, Int, Uint. Very minor nitpick and no impact on correctness though.

Reply:
The reason is that the last case uses `dataCast()`, and issues a warning. I suspect it wouldn't be measurable, but if we cared, we could refactor to `switch( value->typeId() )` and issue the warning in the default case.